gobblin-data-management/src/test/java/org/apache/gobblin/data/management/copy/iceberg/IcebergTableMetadataValidatorTest.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.data.management.copy.iceberg;
+
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.avro.AvroSchemaUtil;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.shaded.org.apache.avro.SchemaBuilder;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class IcebergTableMetadataValidatorTest {
+  private static final org.apache.iceberg.shaded.org.apache.avro.Schema avroDataSchema1 =
+      SchemaBuilder.record("schema1")
+          .fields()
+          .requiredString("field1")
+          .requiredString("field2")
+          .endRecord();
+  private static final org.apache.iceberg.shaded.org.apache.avro.Schema avroDataSchema2 =
+      SchemaBuilder.record("schema2")
+          .fields()
+          .requiredString("field2")
+          .requiredString("field1")
+          .endRecord();
+  private static final org.apache.iceberg.shaded.org.apache.avro.Schema avroDataSchema3 =
+      SchemaBuilder.record("schema3")
+          .fields()
+          .requiredString("field1")
+          .requiredString("field2")
+          .requiredInt("field3")
+          .endRecord();
+  private static final org.apache.iceberg.shaded.org.apache.avro.Schema avroDataSchema4 =
+      SchemaBuilder.record("schema3")
+          .fields()
+          .requiredInt("field1")
+          .requiredString("field2")
+          .requiredInt("field3")
+          .endRecord();
+  private static final Schema schema1 = AvroSchemaUtil.toIceberg(avroDataSchema1);
+  private static final Schema schema2 = AvroSchemaUtil.toIceberg(avroDataSchema2);
```

Comment:
I presume it was the intent... 

would be clearer to name such as, `schema2IsSchema1Compat`

or `schema4IsNotSchema3Compat`